builtin/providers/openstack/resource_openstack_dns_recordset_v2.go

```diff
@@ -134,22 +136,25 @@ func resourceDNSRecordSetV2Read(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error creating OpenStack DNS client: %s", err)
 	}
 
-	zoneID := d.Get("zone_id").(string)
+	// Obtain relevant info from parsing the ID
+	zoneID, recordsetID, err := parseDNSV2RecordSetID(d.Id())
+	if err != nil {
+		return err
+	}
 
-	n, err := recordsets.Get(dnsClient, zoneID, d.Id()).Extract()
+	n, err := recordsets.Get(dnsClient, zoneID, recordsetID).Extract()
 	if err != nil {
 		return CheckDeleted(d, err, "record_set")
 	}
 
-	log.Printf("[DEBUG] Retrieved  record set %s: %#v", d.Id(), n)
+	log.Printf("[DEBUG] Retrieved  record set %s: %#v", recordsetID, n)
 
 	d.Set("name", n.Name)
 	d.Set("description", n.Description)
 	d.Set("ttl", n.TTL)
 	d.Set("type", n.Type)
 	d.Set("records", n.Records)
 	d.Set("region", GetRegion(d))
-	d.Set("zone_id", n.ZoneID)
```

Comment:
:+1:  I was just narrowing down the search space for things that were causing errors. I found it in the test (I hadn't added `parseDNSV2RecordSetID`). Running the tests now to verify they pass.